tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md

```diff
@@ -582,12 +592,70 @@ In some cases, users need to register a lot (10+) of sanitizers. In this case, g
 ]
 ```
 
+### Set a Matcher
+
+Setting a matcher is just like adding a `sanitizer`. Set the `x-abstraction-identifier` value to the name of the matcher you want to instantiate, and provide the proper constructor arguments in the body! Check `Info/Available/` for available matchers.
```

Comment:
when adding a sanitizer. The latter set sanitizer overrides the one sent previously though, right?

Reply:
Nope. Sanitizers can be stacked.

Reply:
A `matcher` needs to take a unit of input, then compare it to all the possible entries in a set.

It does this by walking each possible entry, and comparing URI + Body + Headers. As there are identical values found, an integer is incremented. At the end, the "most matching" entry that has not been disqualified is popped off the stack. (We can't present the same response twice this way).

Knowing that, it becomes quite difficult to figure out how one would combine a matcher.

A `sanitizer` on the other hand, merely run against two targets:

1. Every entry in the recording, prior to saving to disk.
2. Before matching a request to a recording.

We can handle multiple sanitizers by simply running them one after another.